File: c++/6_zigzag_conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if(numRows == 1)
            return s;
        
        vector<string> str(numRows);
        
        bool down = true;
        int i = 0;
        for(int j = 0; j<s.length(); j++){
            while(i != numRows-1 && j<s.length()){
                    char ch = s[j];
                    str[i].push_back(ch);
                    i++;
                    j++;
            }
            
            while(i != 0 && j<s.length()){
                    char ch = s[j];
                    str[i].push_back(ch);
                    i--;
                    j++;
            }
            j--;
        }
        
        string ans;
        for( auto j:str )
            ans += j;
        
        return ans;
    }
};

File: java/6_zigzag_conversion.java
class Solution {
    public String convert(String s, int numRows) {
        String[] ans = new String[numRows];
        int n = s.length() ;
        boolean flag = true;
        int j = 0;
        for(int i = 0; i<n; i++){
            if(flag){
                ans[j] += s[i];
                j++;
                if(j == numRows-1){
                    flag = !flag;
                }
            }
            
            else{
                ans[j] += s[i];
                j--;
                if(j == 0)
                    flag = !flag;
            }
        }
        
        String str = "";
        for(int i = 0; i<numRows; i++)
            str = ans[i];
        
        return str;
    }
}